SCAFFOLDING.md

```diff
@@ -1,99 +1,145 @@
+To create a `scaffold`, you must create a [`yeoman-generator`](http://yeoman.io/authoring/). Because of that, you can optionally extend your generator to include methods from the [Yeoman API](http://yeoman.io/learning/). It's worth noting that we support all the properties of a regular Webpack configuration. In order for us to do this, there's a thing you need to remember:
 
-Objects are made using strings, while strings are made using double strings. This means that in order for you to create an string, you have to wrap it inside another string for us to validate it correctly.
+> Objects are made using strings, while strings are made using double strings. This means that in order for you to create a string, you have to wrap it inside another string for us to validate it correctly.
 
+### Required
+- [opts.env.configuration](#optsenvconfiguration-required)
+- [opts.env.configuration.myObj](#optsenvconfigurationmyObj-required)
+- [myObj.webpackOptions](#myObjwebpackOptions-required)
+- [writing()](#writing()-required)
+  
+### Optional
+- [myObj.merge](#myObjmerge-optional)
+- [myObj.topScope](#myObjtopScope-optional)
+- [myObj.configName](#myObjconfigName-optional)
 
 ### `opts.env.configuration`(required)
 
-Initialized inside the constructor of your generator in order for the CLI to work.
+This is the entry point your configuration, initialize it inside the constructor of your generator in order for the CLI to work:
 
 ```js
 constructor(args, opts) {
-		super(args, opts);
-		opts.env.configuration = {};
-	}
+	super(args, opts);
+	opts.env.configuration = {};
+}
 ```
 ### `opts.env.configuration.myObj` (required)
 
-`myObj` is your scaffold. This is where you will add options for the CLI to transform into a configuration. You can name it anything, and you can also add more objects, that could represent a `dev.config` or `prod.config`.
+This is your scaffold, you add here the options that the CLI will transform into a Webpack configuration. You can have many different scaffolds named as you prefer, representing different configurations like `dev.config` or `prod.config`:
 
 ```js
 constructor(args, opts) {
-		super(args, opts);
-		opts.env.configuration = {
-			dev: {},
-			prod: {}
-		};
-	}
+	super(args, opts);
+	opts.env.configuration = {
+		dev: {},
+		prod: {}
+	};
+}
 ```
 
 ### `myObj.webpackOptions` (required)
 
-As with a regular webpack configuration, this property behaves the same. Inside `webpackOptions` you can declare the properties you want to scaffold. You can for instance, scaffold `entry`, `output` and `context`.
+This object has the same format as a regular Webpack configuration, so you declare here the properties that you want to scaffold, like `entry`, `output` and `context`. You can initialize this inside a yeoman method:
```

Comment:
Please convert `Webpack` to `webpack`.
It is as per our branding guidelines : https://webpack.js.org/branding/

Reply:
Didn't knew that, thanks. Done ;)